climada/engine/forecast.py

```diff
@@ -0,0 +1,897 @@
+    """
+
+    def __init__(self,
+                 hazard_dict,
+                 exposure,
+                 impact_funcs,
+                 haz_model=None,
+                 exposure_name=None):
+        """ Initialization with hazard, exposure and vulnerability.
+
+        Parameters
+        ----------
+        hazard_dict: dict
+            Dictionary of the format {run_datetime: Hazard} with run_datetime
+            being the initialization time of a weather forecast run and Hazard
+            being a CLIMADA Hazard derived from that forecast for one event.
+            A probabilistic representation of that one event is possible,
+            as long as the attribute Hazard.date is the same for all
+            events. Several run_datetime:Hazard combinations for the same
+            event can be provided.
+        exposure: Exposure
+        impact_funcs: ImpactFuncSet
+        haz_model: str, optional
+            Short string specifying the model used to create the hazard,
+            if possible three big letters. Default is 'NWP' for numerical
+            weather prediction.
+        exposure_name: str, optional
+            Short string specifying the exposure, which is used in filenames.
+        """
+        self.run_datetime = [key for key in hazard_dict.keys()]
+        self.hazard = [hazard_dict[key] for key in self.run_datetime]
+        # check event_date
+        hazard_dates = [date for hazard in self.hazard for date in hazard.date]
+        if not len(np.unique(hazard_dates)) == 1:
+            raise ValueError('Please provide hazards containing only one ' +
+                              'event_date. The current hazards contain several ' +
+                              'events with different event_dates and the Forecast ' +
+                              'class cannot function proberly with such hazards.')
+        self.event_date = dt.datetime.fromordinal(np.unique(hazard_dates)[0])
+        if haz_model is None:
+            self.haz_model = 'NWP'
+        else:
+            self.haz_model = haz_model
+        self.exposure = exposure
+        if exposure_name is None:
+            try:
+                self.exposure_name = iso_cntry.get(exposure.gdf.region_id.unique()[0]).name
+            except (KeyError, AttributeError):
+                self.exposure_name = 'custom'
+        else:
+            self.exposure_name = exposure_name
+        self.vulnerability = impact_funcs
+        self._impact = [Impact() for i in range(len(self.run_datetime))]
+
+    def ei_exp(self, run_datetime=None):
+        """
+        Expected annual impact
+
+        Parameters
+        ----------
+        run_datetime: datetime.datetime, optional
+            Select the used hazard by the run_datetime.
+        Returns
+        -------
+        float
+        """
+        if run_datetime is None:
+            run_datetime = self.run_datetime[0]
+        haz_ind = np.argwhere(np.isin(self.run_datetime, run_datetime))[0][0]
```

Comment:
Why not 
```suggestion
        haz_ind = np.argwhere(self.run_datetime == run_datetime).tolist()
```

Reply:
I think your suggestion would return `haz_ind = [[1]]` I need `haz_ind = 1`

Reply:
I would return `haz_ind = [1]` right? Then make `[haz_ind] = np.argwhere(self.run_datetime == run_datetime).tolist()`

The previous version contains an unnecessary call to `np.isin` ...

Reply:
On my system `self.run_datetime == run_datetime` does not do elementwise comparison but just returns `False`